Dockerfile

```diff
@@ -3,21 +3,43 @@ ARG bashver=latest
 FROM bash:${bashver}
 ARG TINI_VERSION=v0.19.0
 ARG TARGETPLATFORM
+ARG TEMPDIR="/tmp/build"
+ARG BATS_LIBS_DESTDIR="/usr/lib"
+ARG BATS_LIBS_SUPPORT_VERSION=0.3.0
+
+# https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL maintainer="Bats-core Team"
+LABEL org.opencontainers.image.authors="Bats-core Team"
+LABEL org.opencontainers.image.title="Bats"
+LABEL org.opencontainers.image.description="Bash Automated Testing System"
+LABEL org.opencontainers.image.url="https://hub.docker.com/r/bats/bats"
+LABEL org.opencontainers.image.source="https://github.com/bats-core/bats-core"
+LABEL org.opencontainers.image.base.name="docker.io/bash"
 
 COPY ./docker /tmp/docker
 # default to amd64 when not running in buildx environment that provides target platform
 RUN /tmp/docker/install_tini.sh "${TARGETPLATFORM-linux/amd64}"
-    
 
 # Install parallel and accept the citation notice (we aren't using this in a
 # context where it make sense to cite GNU Parallel).
 RUN apk add --no-cache parallel ncurses && \
-    mkdir -p ~/.parallel && touch ~/.parallel/will-cite
+    mkdir -p ~/.parallel && touch ~/.parallel/will-cite \
+    && addgroup -S bats \
+    && adduser -S bats -G bats -s "/bin/bash" \
+    && mkdir /code && chown bats:bats /code \
+    # Bats support
+    && mkdir -p ${TEMPDIR}/bats-support ${BATS_LIBS_DEST_DIR}/bats-support/src \
+    && wget -qO- https://github.com/bats-core/bats-support/archive/refs/tags/v${BATS_LIBS_SUPPORT_VERSION}.tar.gz | tar xvz -C ${TEMPDIR}/bats-support --strip-components 1 \
+    && install -Dm755 ${TEMPDIR}/bats-support/load.bash ${BATS_LIBS_DESTDIR}/bats-support/load.bash \
```

Comment:
I think it would be better to have dedicated install scripts in each project and rely on those for the packaging.
However, I am not even sure if we want to use curl here or use the [checkout action](https://github.com/actions/checkout).

Reply:
>I think it would be better to have dedicated install scripts in each project and rely on those for the packaging.
However, I am not even sure if we want to use curl here or use the [checkout action](https://github.com/actions/checkout).

Honestly I prefer to address both of your comments in a separated bash script instead of using the github CI for a very important reason: adding an external ci step (GH actions) as prerequisite for a dockerfile will prevent to build the docker img in people laptop and from any kind other of CI system.

Let my take a tentative to create a shell script so we can make this Dockerfile more elegant and slim, will ping you asap

Reply:
You're right, I should not have mixed two concerns. This idea came from the wish for integration tests. But that should be solved in a different manner.